discovery/spec.md

```diff
@@ -365,775 +244,693 @@ The following sections define the attributes that appear in a Service entity.
     }
     ````
 
-##### specversions
+#### channel
 
-- Type: Array of `String` values
-- Description: CloudEvents
-  [`specversions`](../cloudevents/spec.md#specversion)
-  that can be used for events published for this service.
+- Type: String
+- Description: A string that can be used to correlate Endpoints. Any Endpoints
+  within an instance of a Discovery Service that share the same `channel` value
+  MUST have some relationship. This specification does not define that
+  relationship or the specific values used in this property. 
+
+  This property MAY be serialized as an empty string or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
-  - REQUIRED
-  - MUST be a non-empty array of non-empty strings
+  - OPTIONAL
+  - if present, MUST be a string
+- Examples:
+  - `queue1`
+
+#### authscope
+
+- Type: TBD
+- Description: TBD
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - TBD
 
-##### subscriptionurl
+#### usage
 
-- Type: `URL`
-- Description: An absolute URL indicating where CloudSubscriptions `subscribe`
-  API calls MUST be sent to.
+- Type: String
+- Description: The interaction model supported by this Endpoint. This
+  specification defines a set of possible values (see the
+  [Usage Values](#usage-values) section for more information, however,
+  implementations MAY define new values as extensions. It is RECOMMENDED
+  that extension values be defined with some organizational unique string
+  to reduce the chances of collisions with possible future spec-defined values.
 - Constraints:
   - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - `consumer`
+
+#### config
 
-##### subscriptionconfig
+- Type: Object
+- Description: Metadata the provides additional information concerning the
+  interactions with this Endpoint. Each `usage` value MUST define the set of
+  `config` properties that are valid.
 
-- Type: `Map` of `String` to `String`
-- Description: A map indicating supported options for the `config` parameter for
-  the CloudSubscriptions subscribe() API call. Keys are the name of keys in the
-  allowed config map, the values indicate the type of that parameter, conforming
-  to the CloudEvents
-  [type system](../cloudevents/spec.md#type-system).
-  TODO: Needs resolution with CloudSubscriptions API
+  This property MAY be serialized as an empty object or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
 - Examples:
-  - ??
+  - `{ "protocol": "kafka" , "endpoint": "https://example.com/queue/1234" }`
```

Comment:
I wonder if this example is misleading, as it needs the `usage` to be understood. That might tell me that usage and options should be colocated in an object together one level off of the root?

Is this example of a producer?

Yeah it feels like this would be easier to program to if we had more of this kind of structure:

```
{
   ...
   "usage": {
      "role": "producer",
      "kafka": {
         "queue": "https://example.com/queue/1234"
      }
   }
}
```

No, I am not sure about that... I am thinking about two things: 

1. How would I model this as a protobuf?
2. How do we avoid the downsides of something like OSBAPI where there are generic overloaded fields that  based on some config in the response or context.

Reply:
Let's chat on the call. I don't have a strong opinion on this one as it feels like just a syntax diff to me and not one that impacts the semantics.